recipes/openmpi/all/conanfile.py

```diff
@@ -1,97 +1,264 @@
-    topics = ("conan", "mpi", "openmpi")
     description = "A High Performance Message Passing Library"
     license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.open-mpi.org"
+    topics = ("mpi", "openmpi")
+    provides = "mpi"
+    package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
-        "fortran": ["yes", "mpifh", "usempi", "usempi80", "no"]
+        "fortran": ["yes", "mpifh", "usempi", "usempi80", "no"],
+        "enable_cxx": [True, False],
+        "enable_cxx_exceptions": [True, False],
+        "with_verbs": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
-        "fortran": "no"
+        "fortran": "no",
+        "enable_cxx": False,
+        "enable_cxx_exceptions": False,
+        "with_verbs": False,
     }
 
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        if not self.options.enable_cxx:
+            self.settings.rm_safe("compiler.libcxx")
+            self.settings.rm_safe("compiler.cppstd")
+            del self.options.enable_cxx_exceptions
+        if is_apple_os(self):
+            # Unavailable due to dependency on libnl
+            del self.options.with_verbs
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        # OpenMPI public headers don't include anything besides stddef.h.
+        # transitive_headers=True is not needed for any dependencies.
+        self.requires("hwloc/2.10.0")
+        self.requires("zlib/[>=1.2.11 <2]")
+        if self.settings.os == "Linux":
+            self.requires("libnl/3.8.0")
+        if self.options.get_safe("with_verbs"):
+            self.requires("rdma-core/52.0")
+
+    def validate(self):
         if self.settings.os == "Windows":
+            # Requires Cygwin or WSL
             raise ConanInvalidConfiguration("OpenMPI doesn't support Windows")
 
-    def requirements(self):
-        # FIXME : self.requires("libevent/2.1.12") - try to use libevent from conan
-        self.requires("zlib/1.2.11")
+        if self.version == "4.1.0" and is_apple_os(self) and self.settings.arch == "armv8":
+            # INFO: https://github.com/open-mpi/ompi/issues/8410
+            raise ConanInvalidConfiguration(f"{self.ref} is not supported in Mac M1. Use a newer version.")
+
+        if conan_version.major == 1 and self.settings.compiler in ["clang", "apple-clang"]:
```

Comment:
If the recipe was working with Conan 1 this would be a regression which we will not accept.
Please provide the build logs so that we can investigate

Reply:
https://github.com/conan-io/conan-center-index/pull/18980#issuecomment-2321213466

Reply:
Thanks - we will reproduce this locally to inspect the config.log and find out what is going on.

if the recipe was working before, a migration to conan v2 should not result in this not working anymore, or worse, publishing a new recipe revision for an existing version that goes for working to non-working for Conan 1 uses. This is exposing our users to a disruption that is not acceptable.

if the CI is failing in a way that you are not able to troubleshoot with the information in the build logs, please flag this  so that we can investigate. When there is an team member actively reviewing a PR, we will test this and fix any issues locally and push the fixes to expedite the review and merge.

Reply:
Understandable. I'll try to debug the configure failure as well. It does not reproduce on my Linux system directly, but I managed to reproduce it on a Conan Docker image at least.